tests/test_cli.py

```diff
@@ -1,23 +1,410 @@
+            "--input-external",
+            server_a_verify1_output_filename,
+            "--output",
+            server_b_verify2_path,
+        ],
+    )
+    assert result.exit_code == 0
+
+    _validate_intermediate_output(
+        server_a_verify2_output_filename, server_b_verify2_output_filename
+    )
+
+    ###########################################################
+    # aggregate
+    ###########################################################
+    (server_a_aggregation_path, server_b_aggregation_path) = [
+        os.path.join(path, "intermediate", "external", "aggregate")
+        for path in (server_a_bucket_path, server_b_bucket_path)
+    ]
+    (server_a_aggregation_output_filename, server_b_aggregation_output_filename) = [
+        os.path.join(path, "data.ndjson")
+        for path in (server_a_aggregation_path, server_b_aggregation_path)
+    ]
+
+    # server A
+    os.makedirs(server_a_aggregation_path)
+    result = runner.invoke(
+        commands.aggregate,
+        [
+            "--n-data",
+            len(client_data),
+            "--batch-id",
+            batch_id,
+            "--server-id",
+            "A",
+            "--private-key-hex",
+            keygen_server_a["private_key"],
+            "--shared-secret",
+            shared_seed,
+            "--public-key-hex-internal",
+            keygen_server_a["public_key"],
+            "--public-key-hex-external",
+            keygen_server_b["public_key"],
+            "--input",
+            os.path.join(server_a_share_path, "data.ndjson"),
+            "--input-internal",
+            server_a_verify2_output_filename,
+            "--input-external",
+            server_b_verify2_output_filename,
+            "--output",
+            server_a_aggregation_path,
+        ],
+    )
+    assert result.exit_code == 0
+
+    # server B
+    os.makedirs(server_b_aggregation_path)
+    result = runner.invoke(
+        commands.aggregate,
+        [
+            "--n-data",
+            len(client_data),
+            "--batch-id",
+            batch_id,
+            "--server-id",
+            "B",
+            "--private-key-hex",
+            keygen_server_b["private_key"],
+            "--shared-secret",
+            shared_seed,
+            "--public-key-hex-internal",
+            keygen_server_b["public_key"],
+            "--public-key-hex-external",
+            keygen_server_a["public_key"],
+            "--input",
+            os.path.join(server_b_share_path, "data.ndjson"),
+            "--input-internal",
+            server_b_verify2_output_filename,
+            "--input-external",
+            server_a_verify2_output_filename,
+            "--output",
+            server_b_aggregation_path,
+        ],
+    )
+    assert result.exit_code == 0
+
+    # FIXME: should have some kind of output of the expected output?
+    # For this case it looks like we get results like this: "AJOhAKEAoQA=". Is
+    # that correct?
```

Comment:
@acmiyaguchi do you have any suggestions for this part? this is what I'm seeing looking at the file, but it seems wrong